### ISSUE TITLE
__geo_interface__ and Empty Geometries

### ISSUE THREAD

**Issue by bixb0012** [Toblerity/Shapely#450]
I am not sure if this is an undefined/under-defined part of [`__geo_interface__`](https://gist.github.com/sgillies/2217756) or a bug in Shapely, but I have run into issues with empty polygons and `__geo_interface__`.
```python
>>> import shapely
>>> from shapely import wkt
>>> 
>>> shapely.__version__
'1.5.17'
>>> 
>>> pg = wkt.loads('POLYGON EMPTY')
>>> pg.wkt
'POLYGON EMPTY'
>>> pg.__geo_interface__
Runtime error 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python27\ArcGIS10.5\lib\site-packages\shapely\geometry\polygon.py", line 300, in __geo_interface__
    coords = [tuple(self.exterior.coords)]
AttributeError: 'NoneType' object has no attribute 'coords'
>>> 
```

Thoughts?

**Comment by snorfalorpagus** [comment]
Related to #442 / #403. This should be fixed for Point and LineString. I'm not sure what the current behaviour for an empty Polygon is.

**Comment by bixb0012** [comment]
@snorfalorpagus, thanks for the references, I will take a look.  Obviously my search skills failed me on this one, maybe because I didn't even bother applying them before posting. :blush: 

If the issue is going to be fixed for Point and LineString, it would be good to address Polygon as well.

**Comment by bixb0012** [comment]
Looking at 1.6b3:
```python
>>> import shapely
>>> from shapely import wkt
>>> from shapely import geometry
>>> 
>>> shapely.__version__
'1.6b3'
>>> 
>>> output_str = "{}\nWKT              : {}\n__geo_interface__: {}\n"
>>> for geom_type in geometry.base.GEOMETRY_TYPES:
...     geom = wkt.loads(geom_type + ' EMPTY')
...     try:
...         print output_str.format(geom_type,
...                                 geom.wkt,
...                                 str(geom.__geo_interface__))
...     except Exception as e:
...         print output_str.format(geom_type,
...                                 geom.wkt,
...                                 e)
...  
Point
WKT              : POINT EMPTY
__geo_interface__: list index out of range

LineString
WKT              : LINESTRING EMPTY
__geo_interface__: {'type': 'LineString', 'coordinates': ()}

LinearRing
WKT              : LINEARRING EMPTY
__geo_interface__: {'type': 'LinearRing', 'coordinates': ()}

Polygon
WKT              : POLYGON EMPTY
__geo_interface__: 'NoneType' object has no attribute 'coords'

MultiPoint
WKT              : MULTIPOINT EMPTY
__geo_interface__: {'type': 'MultiPoint', 'coordinates': ()}

MultiLineString
WKT              : MULTILINESTRING EMPTY
__geo_interface__: {'type': 'MultiLineString', 'coordinates': ()}

MultiPolygon
WKT              : MULTIPOLYGON EMPTY
__geo_interface__: {'type': 'MultiPolygon', 'coordinates': []}

GeometryCollection
WKT              : GEOMETRYCOLLECTION EMPTY
__geo_interface__: {'type': 'GeometryCollection', 'geometries': []}

>>> 
```
It appears `__geo_interface__` works with all base geometry types in Shapely except `POINT` and `POLYGON`.  Additionally, the `__geo_interface__` output appears to mix and match empty tuples and lists, or round and square brackets.

**Comment by sgillies** [comment]
Thanks for the report @bixb0012. This is indeed a bug (or bugs).

**Comment by bixb0012** [comment]
I have an idea to pitch for how to address these issues.  I should be able to get a pull request together for people to review either today or tomorrow.

**Comment by bixb0012** [comment]
Pulled it together (no pun intended): #452

**Comment by sgillies** [comment]
The AttributeError is fixed in e4c73f3.
